src/pages/Onboard/NameTeam.vue

```diff
@@ -130,18 +136,92 @@ export default {
       await this.createLicense()
       if (!this.updateServerError) this.goToResources()
     },
-    goToResources() {
+    async accept(pt) {
+      try {
+        this.accepting = true
+        const invitationId = pt.id
+        const accepted = await this.acceptMembershipInvitation(invitationId)
+        this.redirectTenant = pt.tenant.slug
+        if (accepted.accept_membership_invitation.id) {
+          this.redirectTenant = pt.tenant.slug
+          this.accepting = false
+        }
+      } catch (e) {
+        this.mutationErrorMessage = e
+          .toString()
+          .split(':')
+          .pop()
+        this.loading = false
+        this.error = true
+      }
+      this.loading = false
+    },
+    async decline(id) {
+      try {
+        this.deleting = true
+        const declined = await this.declineMembershipInvitation(id)
+        if (declined.delete_membership_invitation.success) {
+          this.$emit('hide')
+        }
+        this.deleting = false
+      } catch (e) {
+        this.mutationErrorMessage = e
+          .toString()
+          .split(':')
+          .pop()
+        this.loading = false
+        this.error = true
+      }
+    },
+    async goToResources() {
       this.revealNameInput = false
       this.revealUrlInput = false
       this.revealConfirm = false
 
+      await this.setCurrentTenant(
+        this.redirectTenant != null
+          ? this.redirectTenant
+          : this.tenantChanges.slug || this.tenant.slug
+      )
+
       this.$router.push({
         name: 'onboard-resources',
         params: { tenant: this.tenant.slug }
       })
     }
   },
-  apollo: {}
+  apollo: {
+    pendingInvitations: {
+      query: require('@/graphql/Tenant/pending-invitations-by-email.gql'),
+      variables() {
+        return {
+          email: this.user.email
+        }
+      },
+      async result({ data, loading }) {
+        if (loading || !data || !data.pendingInvitations) return
+        // We filter this because we don't want to show invitations
+        // to tenants we're already in...
+        // This is due to a bug(feature?) in the back end that allows
+        // users to be invited to tenants they're already part of
+        await this.getUser()
+        this.pendingInvitations =
+          data.pendingInvitations && data.pendingInvitations.length
+            ? data.pendingInvitations.filter(
+                pi =>
+                  !this.memberships
+                    .map(at => at.tenant.id)
+                    .includes(pi.tenant.id)
+              )
+            : []
+      },
+      skip() {
+        return !this.isCloud || !this.user || !this.user.email
+      },
+      fetchPolicy: 'network-only',
+      pollInterval: 60000
+    }
```

Comment:
I know this is from the sidenav component but I think we should re-evaluate this logic in a follow-up PR @zhen0

Reply:
Sorry @znicholasbrown - I missed this when you initially commented.  Want me to do a follow up issue?

Reply:
https://github.com/PrefectHQ/ui/issues/381 👍